session/session.go

```diff
@@ -1303,6 +1303,10 @@ func (s *session) GetGlobalSysVar(name string) (string, error) {
 		return "", variable.ErrUnknownSystemVar.GenWithStackByArgs(name)
 	}
 
+	if sv.HasInstanceScope() { // has INSTANCE scope only, not pure global
+		return "", errors.New("variable has only instance scope and no GetGlobal func. Not sure how to handle yet")
+	}
```

Comment:
I think we can add a test so that all instance scoped variables need a GetGlobal function for now (until we later refactor in stage 3 and make sysvars the single source of truth for configuration).

You can add it to `TestInstanceScope` in `sessionctx/variable/sysvar_test.go`:

```go
// InstanceScoped variables require a Gettter and a Setter for now
// Later we will expand functionality where they might not need this.
for _, sv := range GetSysVars() {
    if sv.HasInstanceScope() {
        require.NotNil(sv.GetGlobal)
        require.NotNil(sv.SetGlobal)
    }
}
```

If this test is in place, you can clean up the errors.New to use be: `errors.New("variable is instance scope")`, since this should be unreachable.